### PR TITLE
Organizations should be selected from the LIST of EXISTING organizations created by THIS user

### LINKED ISSUE #623: Organizations should be selected from the LIST of EXISTING organizations created by THIS user
image

### PR DESCRIPTION
**Pull request by Cheshiriks**
Organizations should be selected from the LIST of EXISTING organizations created by THIS user

What's done:
* changed field from organization
Closes #623 
Сни433

### CODE REVIEW

**Comment by akuleshov7** [comment]
SO FAST???

**Comment by Cheshiriks** [comment]
> SO FAST???

Not all :)
I need this pull request to visit this thread from another computer

**Comment by Cheshiriks** [comment]
> 

Out of habit :(

**Comment by Cheshiriks** [comment]
> > > SO FAST???
> > 
> > 
> > Not all :) I need this pull request to visit this thread from another computer
> 
> Why are you requesting review, if it's not ready for review yet? 😕

Out of habit :(

File: save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/basic/SelectForm.kt [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    "TOO_LONG_FUNCTION",
+    "LongParameterList"
+)
+internal fun RBuilder.selectFormRequired(
+    form: InputTypes,
+    validInput: Boolean,
+    classes: String,
+    text: String,
+    elements: List<String>,
+    initialValue: String?,
```
Comment (petertrr) [inline save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/basic/SelectForm.kt:32]: Is it currently selected item? It should be in the state of this component, since you've already extracted it into a separate function. Otherwise it makes already large state of `ProjectView` even more complex :(

File: save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/ProjectController.kt [hunk 2]
```diff
@@ -155,10 +155,15 @@ class ProjectController(private val projectService: ProjectService,
      * @return response
      */
     @PostMapping("/save")
+    @Suppress("UnsafeCallOnNullableType")
     fun saveProject(@RequestBody newProjectDto: NewProjectDto, authentication: Authentication): ResponseEntity<String> {
         val userId = (authentication.details as AuthenticationDetails).id
+        val organizationNew = organizationService.findByName(newProjectDto.organizationName)
```
[the remark this answers is not in the document]
Comment (Cheshiriks) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/ProjectController.kt:161]: In order to separate the names of the class variable Project and the organization found for it in the next line, where the assignment takes place.
project.apply { organization = organizationNew }

[the remark this answers is not in the document]
Comment (petertrr) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/ProjectController.kt:161]: `project.apply { this.organization = organization }`

File: save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt [hunk 3]
```diff
@@ -25,6 +25,16 @@ internal class OrganizationController(private val organizationService: Organizat
     fun getOrganizationByName(@RequestParam name: String): Organization =
             organizationService.findByName(name) ?: throw NoSuchElementException("Organization with name [$name] was not found.")
 
+    /**
+     * @param authentication an [Authentication] representing an authenticated request
+     * @return list of organization by owner id
+     */
+    @GetMapping("/get/organizations-owner")
```
[the remark this answers is not in the document]
Comment (Cheshiriks) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt]: Renamed

File: save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/ProjectController.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (Cheshiriks) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/ProjectController.kt]: Fixed

File: save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/views/CreationView.kt [hunk 4]
```diff
@@ -86,6 +83,16 @@ external interface ProjectSaveViewState : State {
      * Validation of input fields
      */
     var gitConnectionCheckingStatus: GitConnectionStatusEnum?
+
+    /**
+     * Organizations of user
+     */
+    var organizationsUser: List<Organization>?
```
[the remark this answers is not in the document]
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/views/CreationView.kt]: Renamed

File: save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/views/CreationView.kt [hunk 5]
```diff
@@ -115,16 +122,53 @@ class CreationView : AbstractView<Props, ProjectSaveViewState>(true) {
...
+        target: Event,
+        isProject: Boolean = true,
+        isHtmlInputElement: Boolean = true
+    ) {
+        val tg = if (isHtmlInputElement) {
+            (target.target as HTMLInputElement).value
+        } else {
+            (target.target as HTMLSelectElement).value
+        }
+        if (isProject) fieldsMap[fieldName] = tg else fieldsMap[fieldName] = tg
```
[the remark this answers is not in the document]
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/views/CreationView.kt]: Fixed

File: save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/basic/SelectForm.kt (hunk 1 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/basic/SelectForm.kt]: Fixed

File: save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (petertrr) [reply save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/OrganizationController.kt]: I mean, without `get`. The action is kind of present in HTTP method:
`GET /organizations/list` vs `GET /organizations/get/list`

File: save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/basic/InputForms.kt [hunk 6]
```diff
@@ -17,7 +17,7 @@ import kotlinx.html.js.onChangeFunction
  * @property str
  */
 @Suppress("WRONG_DECLARATIONS_ORDER")
-internal enum class InputTypes(val str: String) {
+enum class InputTypes(val str: String) {
```
Comment (petertrr) [inline save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/basic/InputForms.kt:20]: Why did you remove `internal`?

File: save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/basic/SelectForm.kt [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+
+    /**
+     * elements of `select`
+     */
+    var elements: List<String>?
+
+    /**
+     * initial value of `select`
+     */
+    var initialValue: String?
```
Comment (petertrr) [inline save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/basic/SelectForm.kt:49]: Are these still used? You don't set them from parent, so they're always null?

File: save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/basic/SelectForm.kt [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+                    attrs["class"] = "form-control is-invalid"
+                }
+
+                val newElements = elements?.toMutableList() ?: mutableListOf()
+                newElements.add(0, "")
+                newElements.forEach { element ->
+                    option {
+                        if (element == initialValue) {
+                            attrs.selected = true
+                            setInitialValue(element)
```
Comment (petertrr) [inline save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/basic/SelectForm.kt:111]: Why do you need to set, if they are already equal?

File: save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/views/CreationView.kt [hunk 9]
```diff
@@ -109,22 +107,34 @@ enum class GitConnectionStatusEnum {
...
-    private fun changeFields(fieldName: InputTypes, target: Event, isProject: Boolean = true) {
-        val tg = target.target as HTMLInputElement
-        if (isProject) fieldsMap[fieldName] = tg.value else fieldsMap[fieldName] = tg.value
+    private fun changeFields(
+        fieldName: InputTypes,
+        target: Event,
+        isProject: Boolean = true,
+        isHtmlInputElement: Boolean = true
+    ) {
+        val tg = if (isHtmlInputElement) {
```
Comment (petertrr) [inline save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/views/CreationView.kt:132]: Why not use `when` with `is` checks here?

File: save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/basic/InputForms.kt (hunk 6 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/basic/InputForms.kt:20]: I can't use internal class in external props interface

Comment (petertrr) [reply save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/basic/InputForms.kt:20]: Oh, OK

File: save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/basic/SelectForm.kt (hunk 7 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/basic/SelectForm.kt]: Yes. Removed it

File: save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/basic/SelectForm.kt (hunk 8 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/basic/SelectForm.kt]: Removed it too

File: save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/views/CreationView.kt (hunk 9 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/views/CreationView.kt]: Fixed

File: save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/basic/SelectForm.kt (hunk 7 above)
Comment (petertrr) [reply save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/basic/SelectForm.kt]: And what about `elements`?

File: save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/basic/SelectForm.kt (hunk 8 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/basic/SelectForm.kt]: Yes, it's not needed here.

File: save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/basic/SelectForm.kt (hunk 7 above)
Comment (Cheshiriks) [reply save-frontend/src/main/kotlin/org/cqfn/save/frontend/components/basic/SelectForm.kt]: Fixed

**Review by petertrr — APPROVED** [review]
